### PR TITLE
Documentation: Clarify revisions vs. versions in data model doc

### PR DESCRIPTION
**Pull request by cceckman**
This confused me just looking at the protos; reading up on coreos/etcd#6518 helped clarify things. I highlighted the terms that show up as fields in protos and tried to clean up some of the language.

### CODE REVIEW

File: Documentation/learning/data_model.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ etcd stores data in a multiversion [persistent][persistent-ds] key-value store.
...
 
-A key’s lifetime spans a generation, denoted by its version. Each key may have one or multiple generations. Creating a key increments the version of that key, starting at 1 if the key never existed. Deleting a key generates a key tombstone, concluding the key’s current generation by resetting its version. Each modification of a key increments its version. Once a compaction happens, any version ended before the given revision will be removed and values set before the compaction revision except the latest one will be removed.
+A key's lifetime spans a generation, from creation to deletion. Each key may have one or multiple generations. Creating a key increments the **version** of that key, starting at 1 if the key does not exist at the current revision. Deleting a key generates a key tombstone, concluding the key’s current generation by resetting its version to 0. Each modification of a key increments its version; so, versions are monotonically increasing within a key's generation. Once a compaction happens, any generation ended before the compaction revision will be removed, and values set before the compaction revision except the latest one will be removed.
```
Comment (hexfusion) [inline Documentation/learning/data_model.md]: > A key's lifetime spans a generation, from creation to deletion.

s/lifetime/life?

Comment (cceckman) [reply Documentation/learning/data_model.md]: Sure; done.

("lifetime" was the original; I don't have an opinion one way or another.)

**Comment by cceckman** [comment]
Done (I think- may need to poke Travis after retitling the issue, if I'm reading it right.)

**Comment by hexfusion** [comment]
lgtm
@cceckman, thank you!

**Comment by hexfusion** [comment]
@cceckman Travis retests after new push, all set.

**Pull request merged**